lib/releases.js

```diff
@@ -80,74 +79,84 @@ const contributorsSentence = ({ commits, pullRequests }) => {
 }
 
 const getCategoriesConfig = ({ config }) => {
-  const categoriesConfig = config.categories.reduce((acc, category) => {
-    const labels = category.label ? [category.label] : category.labels
-    labels.forEach(label => (acc[label] = category))
-    return acc
-  }, {})
-
-  const orderedCategories = [
-    UNCATEGORIZED,
-    ..._.flatMap(config.categories, category =>
-      category.label ? category.label : category.labels
-    )
-  ]
-
-  return [categoriesConfig, orderedCategories]
+  const categoriesConfig = [{ title: UNCATEGORIZED }, ...config.categories]
+
+  return categoriesConfig.map(category => {
+    if (!category.labels) category.labels = []
+    if (category.label) {
+      // treat label as labels
+      category.labels.push(category.label)
+      delete category.label
+    }
+    return { ...category, pullRequests: [] }
+  })
 }
 
-const categorizePullRequests = (pullRequests, categories, excludeLabels) => {
-  return pullRequests.reduce(
-    (acc, pullRequest) => {
+const categorizePullRequests = (
+  pullRequests,
+  orderedCategories,
+  excludeLabels
+) => {
+  const categories = orderedCategories.flatMap(category => category.labels)
+
+  // exclude pull requests with exclude labels first than push any uncategorized pull requests to UNCATEGORIZED category
+  const filterPullRequests = pullRequest => {
```

Comment:
The comment describes what the function is doing. I prefer to only have comments to describe the "why". If we encode this in the name of the function, the comment becomes unnecessary. In order to do this we must first make this function responsible for one thing: `filterExcludedPullRequests`. We can create a second function which is responsible for putting uncategorized PRs in a separate array.

Reply:
I'd like to avoid looping over the same thing too many times 😓 But that's just me.